.github/workflows/test.yaml

```diff
@@ -115,9 +123,42 @@ jobs:
         test "$system_daily" == "$current_daily"
       shell: bash
 
+  required-static-checks:
+    runs-on: ubuntu-latest
+    needs:
+      - static-checks
+      - branch-static-checks
+    if: always()
```

Comment:
why do you use always? 
Without always, in case all the dependencies have passed, this should be executed (always green), otherwise it should be skipped. Then if we define as required this step, it will be ok just when all the dependencies have passed. Does it make sense?

Reply:
You are right, skipped will still result in required rule not satisfied.

I went for always() behaviour to make it simple for users, less familiar with the workflow/dependencies, to be able to understand from looking at the test output itself, the exact reason for the failure.